packages/flutter_tools/lib/src/project.dart

```diff
@@ -73,84 +82,59 @@ class FlutterProject {
 
   /// The Android sub project of this project.
-  AndroidProject get android {
-    if (manifest.isModule) {
-      return new AndroidProject(directory.childDirectory('.android'));
-    }
-    return new AndroidProject(directory.childDirectory('android'));
-  }
-
-  /// The generated AndroidModule sub project of this module project.
-  AndroidModuleProject get androidModule => new AndroidModuleProject(directory.childDirectory('.android'));
-
-  /// The generated IosModule sub project of this module project.
-  IosModuleProject get iosModule => new IosModuleProject(directory.childDirectory('.ios'));
-
-  File get androidLocalPropertiesFile {
-    return directory
-        .childDirectory(manifest.isModule ? '.android' : 'android')
-        .childFile('local.properties');
-  }
-
-  File get generatedXcodePropertiesFile {
-    return directory
-        .childDirectory(manifest.isModule ? '.ios' : 'ios')
-        .childDirectory('Flutter')
-        .childFile('Generated.xcconfig');
-  }
+  AndroidProject get android => new AndroidProject._(_androidDirectory, isModule);
 
   File get flutterPluginsFile => directory.childFile('.flutter-plugins');
 
-  Directory get androidPluginRegistrantHost {
-    return manifest.isModule
-        ? directory.childDirectory('.android').childDirectory('Flutter')
-        : directory.childDirectory('android').childDirectory('app');
-  }
-
-  Directory get iosPluginRegistrantHost {
-    // In a module create the GeneratedPluginRegistrant as a pod to be included
-    // from a hosting app.
-    // For a non-module create the GeneratedPluginRegistrant as source files
-    // directly in the iOS project.
-    return manifest.isModule
-        ? directory.childDirectory('.ios').childDirectory('Flutter').childDirectory('FlutterPluginRegistrant')
-        : directory.childDirectory('ios').childDirectory('Runner');
-  }
-
   /// The example sub-project of this project.
-  FlutterProject get example => new FlutterProject._(_exampleDirectory, _exampleManifest, FlutterManifest.empty());
+  FlutterProject get example => new FlutterProject(
+    _exampleDirectory(directory),
+    _exampleManifest,
+    FlutterManifest.empty(),
+  );
+
+  bool get isModule => manifest != null && manifest.isModule;
 
-  /// True, if this project has an example application
-  bool get hasExampleApp => _exampleDirectory.childFile('pubspec.yaml').existsSync();
+  /// True, if this project has an example application.
+  bool get hasExampleApp => _exampleDirectory(directory).existsSync();
 
   /// The directory that will contain the example if an example exists.
-  Directory get _exampleDirectory => directory.childDirectory('example');
+  static Directory _exampleDirectory(Directory directory) => directory.childDirectory('example');
+
+  /// The directory that contains the Android app to be launched by `flutter run`.
+  Directory get _androidDirectory => directory.childDirectory(isModule ? '.android' : 'android');
```

Comment:
What about moving this logic to the AndroidProject class? - it is anyway passed isModule.

Reply:
`AndroidProject` and `IosProject` now take the parent `FlutterProject` as a constructor parameter instead.